File: android-nosql/src/main/java/com/github/florent37/androidnosql/datasaver/LogDataSaver.java
package com.github.florent37.androidnosql.datasaver;

import android.util.Log;

import java.util.Set;

/**
 * Created by florentchampigny on 24/05/2017.
 */

public class LogDataSaver implements DataSaver {

    private final String tag;

    public LogDataSaver(String tag) {
        this.tag = tag;
    }

    @Override
    public void saveNodes(String completePath, Set<String> value) {
        Log.d(tag, "nodes: " + completePath + " -> " + value.toString());
    }

    @Override
    public void saveValue(String completePath, Object value) {
        Log.d(tag, "values: " + completePath + " -> " + value.toString());
    }

    @Override
    public Set<String> getNodes() {
        return null;
    }

    @Override
    public Object getValue(String completePath) {
        return null;
    }

    @Override
    public void remove(String startingPath) {
        Log.d(tag, "remove starting with: " + startingPath);
    }
}

File: android-nosql/src/main/java/com/github/florent37/androidnosql/datasaver/LogDataSaver.kt
package com.github.florent37.androidnosql.datasaver

import android.util.Log

class LogDataSaver(private val tag: String) : DataSaver {

    override val nodes = HashSet<String>()

    override suspend fun saveNodes(completePath: String, value: Set<String>?) {
        Log.d(tag, "nodes: " + completePath + " -> " + value.toString())
    }

    override suspend fun saveValue(completePath: String, value: Any?) {
        Log.d(tag, "values: " + completePath + " -> " + value.toString())
    }

    override suspend fun getValue(completePath: String): Any? {
        return null
    }

    override suspend fun remove(startingPath: String) {
        Log.d(tag, "remove starting with: $startingPath")
    }

}